jetbrains-core/src/software/aws/toolkits/jetbrains/ui/connection/AwsConnectionSettingsSelector.kt

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui.connection
+
+import com.intellij.openapi.project.Project
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.jetbrains.core.credentials.CredentialManager
+import software.aws.toolkits.jetbrains.core.credentials.ProjectAccountSettingsManager
+import software.aws.toolkits.jetbrains.core.region.AwsRegionProvider
+import javax.swing.JComponent
+
+class AwsConnectionSettingsSelector(
+    private val project: Project,
+    private val settingsChangedListener: (AwsRegion?, String?) -> Unit = { _, _ -> }
+) {
+    private val regionProvider = AwsRegionProvider.getInstance()
+    private val credentialManager = CredentialManager.getInstance()
+    private val view = AwsConnectionSettings()
+
+    init {
+        view.region.setRegions(regionProvider.regions().values.toMutableList())
+        view.credentialProvider.setCredentialsProviders(credentialManager.getCredentialProviders())
+
+        val accountSettingsManager = ProjectAccountSettingsManager.getInstance(project)
+        view.region.selectedRegion = accountSettingsManager.activeRegion
+        if (accountSettingsManager.hasActiveCredentials()) {
+            view.credentialProvider.setSelectedCredentialsProvider(accountSettingsManager.activeCredentialProvider)
+        }
+        view.region.addActionListener {
+            fireChange()
+        }
+        view.credentialProvider.addActionListener {
+            fireChange()
+        }
+    }
+
+    private fun fireChange() {
+        settingsChangedListener(view.region.selectedRegion, view.credentialProvider.getSelectedCredentialsProvider())
+    }
+
+    fun selectorPanel(): JComponent = view.panel
+
+    fun resetAwsConnectionOptions(awsConnectionOptions: BaseAwsConnectionOptions) {
```

Comment:
I'd prolly just make this accept `Region` / `credentials` instead of leaking the persistence object.

One of these days I'll get around to introducing an `AwsConnection` object that holds both since it's extremely common to use both together.

Reply:
Makes sense, can change